x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -622,6 +627,38 @@ protected LogicalPlan rule(LogicalPlan e) {
         }
     }
 
+    static class ReplaceMinMaxWithTopHits extends OptimizerRule<Aggregate> {
+
+        @Override
+        protected LogicalPlan rule(Aggregate aggregate) {
```

Comment:
When rewriting an expression, its id needs taking into account:
`SELECT MAX(keyword) AS m FROM x GROUP BY c ORDER BY m`
max has id: 1, get written to First with id: 5 and ORDER BY m (which points to id 1) becomes invalid since that expression id is not available anymore.